lib/events.js

```diff
@@ -621,22 +629,46 @@ function unwrapListeners(arr) {
   return ret;
 }
 
-function once(emitter, name) {
+function once(emitter, name, options = {}) {
+  const { signal } = { ...options };
```

Comment:
Is `{ ...options }` only to handle explicit non-objects here? Can we just omit handling that since there was no third argument anyway?
Also, perhaps we should just use `options ? options.signal : undefined`?